lib/check_runner.rb

```diff
@@ -1,24 +1,26 @@
 class CheckRunner
-  def initialize(*check_names)
+  def initialize(env, *check_names)
     Thread.abort_on_exception = true
 
-    publishing_api_url = ENV["PUBLISHING_API_URL"] || 'http://publishing-api.dev.gov.uk/v2/grouped_content_and_links'
-    checker_db_name = ENV["CHECKER_DB_NAME"] || CheckerDB.in_memory_db_name
-    skip_import = ENV["SKIP_DATA_IMPORT"] ? true : false
-    whitelist_file = ENV["WHITELIST_FILE"] || 'whitelist.yml'
+    publishing_api_url = env["PUBLISHING_API_URL"] || 'http://publishing-api.dev.gov.uk/v2/grouped_content_and_links'
+    checker_db_name = env["CHECKER_DB_NAME"] || CheckerDB.in_memory_db_name
+    skip_import = env["SKIP_DATA_IMPORT"] ? true : false
+    whitelist_file = env["WHITELIST_FILE"] || 'whitelist.yml'
+    @output_dir = env["CHECK_OUTPUT_DIR"] || '.'
+    suppress_progress = env["SUPPRESS_PROGRESS"] ? true : false
 
     checker_db = CheckerDB.new(checker_db_name)
     whitelist = Whitelist.load(whitelist_file)
 
-    progress_reporter = ProgressReporter.new
+    @progress_reporter = suppress_progress ? ProgressReporter.noop : ProgressReporter.new
```

Comment:
Usually it is a good idea to do all the config and wiring in one place, but I don't feel too strongly about it.